pkg/pmem-csi-driver/nodeserver.go

```diff
@@ -798,6 +803,10 @@ func findMountFlags(flags []string, findIn []string) bool {
 		}
 		found := false
 		for _, fIn := range findIn {
+			if f == "dax" && fIn == "dax=always" {
```

Comment:
In current version, instead of adding dax=always, I reverted to adding plain dax, and this check. This is more conservative, however will fail if dax gets deprecated, but that is trouble for the more distant future

Reply:
> This is more conservative, however will fail if dax gets deprecated, but that is trouble for the more distant future

Where exactly is it deprecated? In the "mount" command or in the Linux kernel? We control the former, so as soon as we update the mount util and it breaks, we can change our code without affecting users.

Reply:
Possible future deprecation is mentioned in [kernel doc](https://www.kernel.org/doc/Documentation/filesystems/dax.txt) which means the plan is probably about kernel. mount command has then two possibilities, either remove -o dax support or make it permanent alias to -o dax=always

Reply:
My understanding is the opposite: it describes ""-o dax", which is a mount command line option.

Reply:
The kernel doesn't use "dax" or "dax=always", so this has to be the command line tool.

Reply:
sure, we use mount command API, so we depend what is supported by it. Just that the page I mentioned, explains same concept using kernel flags, and I understand they mention deprecating certain use of flags. So it is probably combined change of mount and kernel which will change it. In all ways, we have to react if something stop working. Our current proposal here is backwards-looking, keeping the syntax which does work with older systems.

Reply:
> Our current proposal here is backwards-looking, keeping the syntax which does work with older systems.

That's my point: how we call the mount command depends on the content of our container, not the system we run on. We can use "dax=always" already now, as our "mount" command understands it.

Where we need to be careful is in what we get back from the kernel. That may show "dax" (old kernel) or "dax=always" (newer kernel), so we have to accept both.